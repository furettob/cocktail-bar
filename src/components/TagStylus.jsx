import * as React from "react"
import './TagStylus.styl';


function TagStylus({ name, type, icon, selected, big, clickCallback }) {

  /* ESE-1 State: add state to manage selection */

  return (
    <span
      className={
        "cb-st-tag" +
        (type?.className ? " cb-st-tag--" + type.className : "") +
        (type?.decoration ? " cb-st-tag--" + type.decoration : "") +
        (selected ? " cb-st-tag--selected" : "") +
        (big ? " cb-st-tag--big" : "")
      }
      onClick={ clickCallback && typeof clickCallback === 'function' ? () => {
        clickCallback()
      } : null}
    >
      {icon && (
        <span>
          <i className={"fa " + icon} />
          &nbsp;
        </span>
      )}
      {name}
    </span>
  )
}

export default TagStylus
