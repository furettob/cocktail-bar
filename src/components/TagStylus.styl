/* TAG */
@require '../style.styl'

.cb-st-tag {
  font-size: $cb-copy-size--small;
  background-color: #cfe9fe;
  border: 1px solid;
  border-color: #034d86;
  color: #034d86;
  border-radius: 100px;
  margin: 0 8px 8px 0;
  padding: 4px 8px;
  white-space: nowrap;
  display: inline-block;
  &--big {
    font-size: 16px;
    margin: 0 12px 12px 0;
    padding: 8px 16px;
  }
}
.cb-st-tag--invertedw {
  background-color: transparent;
  border: 1px solid;
  border-color: #fff;
  color: #fff;
}
.cb-st-tag--selected {
  font-weight: bold;
}
.cb-st-tag--warning {
  background-color: #fff4cc;
  color: #b87d00;
  border-color: #e9a001;
}
.cb-st-tag--success {
  background-color: #ccfadd;
  color: #0a6a36;
  border-color: #0a6a36;
}
.cb-st-tag--disabled {
  background-color: #f7f8fa;
  color: #5d6974;
  border-color: #5d6974;
}
