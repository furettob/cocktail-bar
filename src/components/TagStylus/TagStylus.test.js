import renderer from 'react-test-renderer'
import TagStylus from './TagStylus'

it('renders correctly', () => {
  const tree = renderer
    .create(<TagStylus name="test name"/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
