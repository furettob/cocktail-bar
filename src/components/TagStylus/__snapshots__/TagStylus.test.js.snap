// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="cb-st-tag"
  onClick={null}
>
  I'm testing
  Â 
</span>
`;
